docs/proposals/fast-sync/trust.md

```diff
@@ -0,0 +1,499 @@
+signatures and hash at most 4320 public keys. For one year it's needed
+to check 630720 signatures and compute 315360 hashes. These
+computations can be performed in less than a second. So the main
+property is the size of data which should be downloaded.
+
+### Possible optimizations
+
+Using some optimizations it's possible to improve some properties at
+the cost of making the design and/or implementation a bit more
+complex. With currently proposed scheme the size of a tiny block in
+the worst case scenario is about **1275** kilobytes
+
+* More compact serialization. Instead of using CBOR for everything, we
+  can serialize tiny blocks more compactly. Then a hash will take 28
+  bytes, a simple signature and a public key will take 64 bytes. It
+  can make a tiny block a bit smaller, approximately **1231**
+  kilobytes (**90** Mb per year).
+* Put less signatures into a tiny block. For instance, if all slots
+  are filled, it's enough to put first `k` signatures into a tiny
+  block. If some slots don't have blocks, it's probably not enough to
+  put only `k` signatures, but `2 · l` is not necessary. More profound
+  security analysis should be done to determine lower and upper
+  boundaries. In the best case, if `k` signatures is always enough and
+  the previous optimization is done, the size of a tiny block can be
+  about **674** kilobytes (**49** Mb per year).
+* Another option is to make `l` much smaller than `k`. It depends on
+  how much trust we want to have. `k = 2160` is quite a lot. It's much
+  greater than number of confirmations after a transaction is
+  considered stable. Probably having `l = 500` will be enough for
+  regular end users. In this case the size of a tiny block will be
+  only **284** kilobytes (**21** Mb per year).
+* Compression. Most likely some stakeholders will be slot leaders more
+  than once. In this case it's not necessary to put their public keys,
+  signatures and identifiers more than once. A simple compression can
+  significantly decrease the average size of a tiny block. It's hard
+  to predict exact ratio, but let's assume after compression a block
+  will be 15% smaller. It will require a bit more local computations,
+  but time spent on them should still be negligible. With previous
+  optimizations one tiny block will take about **573** kilobytes if `l
+  ≈ 2160` (**42** Mb per year) or **241** kilobytes if `l = 500` (18
+  Mb per year).
+
+If we get rid of lightweight delegation (which might happen, but is
+not decided yet), the size of a tiny block in the worst case will be
+approximately 1.8 times smaller. It should be less than 10 megabytes
+of data to download for one-year-long blockchain if `l = 500`.
+
+All these optimizations are relatively easy to do, but some of them
+require more research or involve a certain trade-off. We can go even
+further with a bit more complicated optimizations.
+
+* Aggregate signature. Instead of putting `2 · l` or approximately `k`
```

Comment:
You didn't mention another straightforward optimization. Ed25519 allows you to have public keys, SIGs just 32 byte. We have 64kb just because implementation is slightly different. It sure can be adjusted

Reply:
I googled a bit and I think that signatures are 64B in standard Ed25519. Public keys are 32 bytes indeed, I will mention that.